### ISSUE TITLE
Bar plot / bar-width is fixed --> overlap

### ISSUE THREAD

**Issue by engineeringstatistician** [matlab2tikz/matlab2tikz#696]
Hi,

when I create a bar plot in MATLAB, .e.g

```
Y = randn(100,1),
[N, x] = hist(Y, 15);
bar(x,N);
matlab2tikz('test_hist.tikz', 'parseStrings', false, 'ExtraAxisOptions', 'xlabel near ticks, ylabel near ticks', 'height', '\figureheight', 'width', '\figurewidth', 'showInfo', false)
```

then the bars in the compiled .pdf will overlap or are very thin, depending on the figure size and how they are spaced. I think this is because in the generated .tikz it reads

```
\addplot[ybar,bar width=0.8, <...>
```
and thus there is no link to the \figurewidth. This was the case in my previously generated figures using v.0.6.0.

Am I doing something wrong? I could not reproduce this error if I use hist() to directly create the plot, however, I need the indirect way in order to renormalise the histogram data (N) before plotting.

**Comment by egeerardyn** [comment]
The `bar width=0.8` specifies the bar width in axis units, that seems like the most natural thing to do.
However, it seems that the conversion is not happening well, since the `0.8` is wrong from the graphical output.

Thank you for reporting this!

**Comment by engineeringstatistician** [comment]
Awesome that you could verify the bug so quickly.

Besides that it worked flawlessly in v0.6.0 I noticed that there is a difference in the output between bar() und Hist() commands, don't know if that is intentional.

**Comment by egeerardyn** [comment]
The difference in output is not really intentional, but rather an artefact of how `hist` internally calls `bar` in MATLAB. It uses some legacy version of `bar` which we cannot distinguish from a plot with just some random rectangles (`patch`es) in it. I think that's also one of the reasons that The Mathworks has deprecated `hist` in favor for `Histogram` in R2014b/R2015a. Long story short: support for `hist` is low priority (`bar` on the other hand, is still on our radar).

From 0.6 to 1.0, indeed there is a change in how the bar width is converted, and that is obviously bugged.

**Comment by egeerardyn** [comment]
Fixed by #727, this is what I get on R2014b:
![test_hist](https://cloud.githubusercontent.com/assets/177360/9268695/1466acac-4262-11e5-9495-a169725119bb.png)
